mdstat.c

```diff
@@ -110,6 +110,28 @@ static int add_member_devname(struct dev_member **m, char *name)
 	return 1;
 }
 
+/* Detach element from the list, it may modify list_head */
+static void mdstat_ent_list_detach_element(struct mdstat_ent **list_head, struct mdstat_ent *el)
+{
+	struct mdstat_ent *ent = *list_head;
+
+	if (ent == el) {
+		*list_head = ent->next;
+		ent->next = NULL;
+		return;
+	}
+
+	while (ent) {
+		if (ent->next == el) {
+			ent->next = el->next;
```

Comment:
Shouldn't this be assigned to `*list_head`?

Reply:
No, first element has special handling.